File: packages/core/android/src/main/kotlin/com/superapp_sdk/mini_app/MiniAppBridge.kt
package com.superapp_sdk.mini_app

import io.flutter.plugin.common.MethodChannel.Result

class MiniAppBridge private constructor() {
    private var resultCallback: Result? = null

    companion object {
        private val instance = MiniAppBridge()

        fun getInstance(): MiniAppBridge {
            return instance
        }
    }

    fun setResultCallback(callback: Result?) {
        resultCallback = callback
    }

    fun sendResult(resultData: Map<String, Any?>?) {
        resultCallback?.success(resultData)
        resultCallback = null
    }

    fun sendError(errorCode: String, errorMessage: String) {
        resultCallback?.error(errorCode, errorMessage, null)
        resultCallback = null
    }
}

File: packages/core/ios/Classes/mini_app/MiniAppBridge.swift
import Foundation

class MiniAppBridge {
    static let shared = MiniAppBridge()

    private var resultCallback: ((Any?) -> Void)?

    private init() {}

    func setResultCallback(_ callback: @escaping (Any?) -> Void) {
        self.resultCallback = callback
    }

    func finishWithResult(_ result: Any?) {
        resultCallback?(result)
        resultCallback = nil
    }
}